python/lsst/ts/externalscripts/auxtel/latiss_take_twilight_flats.py

```diff
@@ -0,0 +1,167 @@
+
+        self.atcs = None
+        self.latiss = None
+
+    @property
+    def tcs(self):
+        return self.atcs
+
+    @property
+    def camera(self):
+        return self.latiss
+
+    async def configure_tcs(self) -> None:
+        """Handle creating the ATCS object and waiting remote to start."""
+        if self.atcs is None:
+            self.log.debug("Creating ATCS.")
+            self.atcs = ATCS(
+                domain=self.domain,
+                log=self.log,
+            )
+            await self.atcs.start_task
+        else:
+            self.log.debug("ATCS already defined, skipping.")
+
+    async def configure_camera(self) -> None:
+        """Handle creating the camera object and waiting remote to start."""
+        if self.latiss is None:
+            self.log.debug("Creating Camera.")
+            self.latiss = LATISS(
+                self.domain,
+                intended_usage=LATISSUsages.TakeImage,
+                log=self.log,
+                tcs_ready_to_take_data=self.tcs.ready_to_take_data,
+            )
+            await self.latiss.start_task
+        else:
+            self.log.debug("Camera already defined, skipping.")
+
+    @classmethod
+    def get_schema(cls):
+        schema_yaml = """
+            $schema: http://json-schema.org/draft-07/schema#
+            $id: https://github.com/lsst-ts/ts_externalscripts/take_twilight_flats_latiss.yaml
+            title: TakeTwilightFlatsLatiss v1
+            description: Configuration for TakeTwilightFlatsLatiss.
+            type: object
+            properties:
+              filter:
+                description: Filter name or ID.
+                anyOf:
+                  - type: string
+                  - type: integer
+                    minimum: 1
+                  - type: "null"
+              grating:
+                description: Grating name; if omitted the grating is not changed.
+                anyOf:
+                  - type: string
+                  - type: integer
+                    minimum: 1
+                  - type: "null"
+                default: "empty_1"
+            required: ["filter"]
+            additionalProperties: false
+        """
+        schema_dict = yaml.safe_load(schema_yaml)
+
+        base_schema_dict = super(TakeTwilightFlatsLatiss, cls).get_schema()
+
+        for prop in base_schema_dict["properties"]:
+            schema_dict["properties"][prop] = base_schema_dict["properties"][prop]
+
+        return schema_dict
+
+    async def get_sky_counts(self) -> float:
+        """Abstract method to get the median sky counts from the last image.
+
+        Returns
+        -------
+        float
+            Sky counts in electrons.
+        """
+        timeout = 30
+        query = f"SELECT * from cbd_latiss.visit1_quicklook where exposure_id = {self.latest_exposure_id}"
+        item = "post_isr_pixel_median"
+        sky_counts = self.client.wait_for_item_in_row(query, item, timeout)
```

Comment:
Since this is a regular method and not a coroutine, you will need to run this in an executor. It should be something like this:

First, at the top:

```
import functools
```

then here you would do:

```
get_sky_counts = functools.partial(self.client.wait_for_item_in_row, query, item, timeout)
sky_counts = await asyncio.get_running_loop().run_in_executor(
    None, get_sky_counts
)
```

you might want to find the name of the parameters you are passing to `wait_for_item_in_row` and use them instead of relying on their ordering, e.g.:

```
get_sky_counts = functools.partial(self.client.wait_for_item_in_row, query=query, item=item, timeout=timeout)
```

note that I don't know if the names are correct, I just guessing.